### PR TITLE
[AES-CBC] Added more Unit-Tests 

### PR DESCRIPTION
**Pull request by starsheriff**
## Description

Added a couple of tests to test and document the behaviour of the CBC functions in `core/crypto/primitives/aes.go`. Especially two edge cases when a message is encrypted with `CBCEncrypt`  attempted to be decrypted with `CBCPKCS7Decrypt` and vice versa. The work has been done with the assistance of @JonathanLevi. 
## Motivation and Context

We want to add more unit tests to lock/document and verify their correct/expected/standard/compliant behavior.

This is part of #2127 Add a family of AES-CBC related tests
## How Has This Been Tested?

N/A. Should be covered in the future by make unit-test
## Checklist:

- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
- [x] I have either added documentation to cover my changes or this change requires no new documentation.
- [x] I have either added unit tests to cover my changes or this change requires no new tests.
- [x] I have run [golint](https://github.com/golang/lint) and have fixed valid warnings in code I have added or modified. This tool generates false positives so you may choose to ignore some warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Johannes Weissmann jo@starsheriff.eu

### CODE REVIEW

File: core/crypto/primitives/aes_test.go [hunk 1]
```diff
@@ -199,3 +199,249 @@ func TestPKCS7UnPadding(t *testing.T) {
...
+	decrypted, dErr := primitives.CBCDecrypt(key, encrypted)
+
+	if dErr != nil {
+		t.Fatalf("Error encrypting message %v, %v", dErr, decrypted)
+	}
+
+	if string(msg[:]) != string(decrypted[:aes.BlockSize]) {
+		t.Log("msg: ", msg)
+		t.Log("decrypted: ", decrypted[:aes.BlockSize])
+		t.Fatalf("Encryption->Decryption with same key should result in original message")
```
Comment (srderson) [inline core/crypto/primitives/aes_test.go]: Can use `t.Fatal` instead

File: core/crypto/primitives/aes_test.go [hunk 2]
```diff
@@ -199,3 +199,249 @@ func TestPKCS7UnPadding(t *testing.T) {
...
+
+	cipher1, err := primitives.CBCPKCS7Encrypt(key, msg)
+	if err != nil {
+		t.Fatalf("Error encrypting the message.")
+	}
+
+	// expecting a different IV if same message is encrypted with same key
+	cipher2, err := primitives.CBCPKCS7Encrypt(key, msg)
+	if err != nil {
+		t.Fatalf("Error encrypting the message.")
```
Comment (srderson) [inline core/crypto/primitives/aes_test.go]: Can use `t.Fatal` instead.

File: core/crypto/primitives/aes_test.go [hunk 3]
```diff
@@ -199,3 +199,249 @@ func TestPKCS7UnPadding(t *testing.T) {
...
+	copy(decryptionKey, key[:])
+	decryptionKey[0] = key[0] + 1
+
+	var msg = []byte("a message to be encrypted")
+
+	encrypted, _ := primitives.CBCEncrypt(key, msg)
+	decrypted, _ := primitives.CBCDecrypt(decryptionKey, encrypted)
+
+	if string(msg[:]) == string(decrypted[:]) {
+		t.Fatalf("Encryption->Decryption with different keys shouldn't return original message")
```
Comment (srderson) [inline core/crypto/primitives/aes_test.go]: Can use `t.Fatal` instead

File: core/crypto/primitives/aes_test.go [hunk 4]
```diff
@@ -199,3 +199,249 @@ func TestPKCS7UnPadding(t *testing.T) {
...
+	}
+
+	decrypted, dErr := primitives.CBCDecrypt(key, encrypted)
+
+	if dErr != nil {
+		t.Fatalf("Error encrypting message %v", dErr)
+	}
+
+	if string(msg[:]) != string(decrypted[:]) {
+		t.Fatalf("Encryption->Decryption with same key should result in original message")
```
Comment (srderson) [inline core/crypto/primitives/aes_test.go]: Can use `t.Fatal` instead

**Comment by srderson** [comment]
Nice, always good to see more unit tests. Thanks! Added a few comments.

**Comment by JonathanLevi** [comment]
Yes, great work @starsheriff indeed. @srderson: To bridge the time-zone difference (between here and Europe) I quickly pushed a minor update as per the above suggestion. Thanks again.

**Comment by JonathanLevi** [comment]
p.s. Fixes #2166

**Comment by srderson** [comment]
Thanks, LGTM
